// @api3/contracts/dist/src/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.viemConfig = exports.hardhatConfig = exports.CHAINS = exports.DAPPS = exports.deploymentAddresses = exports.dapiManagementMetadata = exports.auctioneerMetadata = void 0;
__exportStar(require("../typechain-types"), exports);
const auctioneer_metadata_json_1 = __importDefault(require("../data/auctioneer-metadata.json"));
exports.auctioneerMetadata = auctioneer_metadata_json_1.default;
const dapi_management_metadata_json_1 = __importDefault(require("../data/dapi-management-metadata.json"));
exports.dapiManagementMetadata = dapi_management_metadata_json_1.default;
const addresses_json_1 = __importDefault(require("../deployments/addresses.json"));
exports.deploymentAddresses = addresses_json_1.default;
__exportStar(require("./proxy"), exports);
var dapps_1 = require("./generated/dapps");
Object.defineProperty(exports, "DAPPS", { enumerable: true, get: function () { return dapps_1.DAPPS; } });
var chains_1 = require("./generated/chains");
Object.defineProperty(exports, "CHAINS", { enumerable: true, get: function () { return chains_1.CHAINS; } });
exports.hardhatConfig = __importStar(require("./hardhat-config"));
exports.viemConfig = __importStar(require("./viem-config"));
__exportStar(require("./types"), exports);
