// @api3/contracts/dist/typechain-types/factories/api3-server-v1/interfaces/IApi3ServerV1OevExtensionOevBidPayer__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IApi3ServerV1OevExtensionOevBidPayer__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "uint256",
                name: "bidAmount",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "data",
                type: "bytes",
            },
        ],
        name: "onOevBidPayment",
        outputs: [
            {
                internalType: "bytes32",
                name: "oevBidPaymentCallbackSuccess",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IApi3ServerV1OevExtensionOevBidPayer__factory {
    static abi = _abi;
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IApi3ServerV1OevExtensionOevBidPayer__factory = IApi3ServerV1OevExtensionOevBidPayer__factory;
