// @api3/contracts/dist/typechain-types/factories/vendor/@openzeppelin/contracts@5.0.2/utils/cryptography/MerkleProof__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerkleProof__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "MerkleProofInvalidMultiproof",
        type: "error",
    },
];
const _bytecode = "0x60566037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea2646970667358221220070ae5905416e1da8aa7f8b0ec0786cb86c056448c2e8151910eb6a30859ba9664736f6c634300081b0033";
const isSuperArgs = (xs) => xs.length > 1;
class MerkleProof__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    connect(runner) {
        return super.connect(runner);
    }
    static bytecode = _bytecode;
    static abi = _abi;
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.MerkleProof__factory = MerkleProof__factory;
